### PR TITLE
Replace MAX_PATH defines with new defines in pal.

### PR DESCRIPTION
**Pull request by Priya91**
Not all references under tests are replaced.

cc @JeremyKuhne @jkotas

### CODE REVIEW

File: src/pal/inc/rt/sal.h [hunk 1]
```diff
@@ -127,7 +127,7 @@
 
    The success criteria can be specified with the _Success_(expr) annotation:
      _Success_(return != FALSE) BOOL
-     PathCanonicalizeA(_Out_writes_(MAX_PATH) LPSTR pszBuf, LPCSTR pszPath) :
+     PathCanonicalizeA(_Out_writes_(MAX_LONGPATH) LPSTR pszBuf, LPCSTR pszPath) :
```
Comment (jkotas) [inline src/pal/inc/rt/sal.h]: The changes in this file should be reverted.

File: src/pal/inc/rt/specstrings_strict.h [hunk 2]
```diff
@@ -609,7 +609,7 @@
 *  Advanced Annotation Examples
 * 
 *  __success(return == TRUE) LWSTDAPI_(BOOL) 
-*  PathCanonicalizeA(__out_ecount(MAX_PATH) LPSTR pszBuf, LPCSTR pszPath);
+*  PathCanonicalizeA(__out_ecount(MAX_LONGPATH) LPSTR pszBuf, LPCSTR pszPath);
```
Comment (jkotas) [inline src/pal/inc/rt/specstrings_strict.h]: Same here.

File: src/pal/inc/rt/sal.h (hunk 1 above)
Comment (Priya91) [inline src/pal/inc/rt/sal.h]: Sure. but why though?

Comment (jkotas) [inline src/pal/inc/rt/sal.h]: PathCanonicalize is defined is hard-coded to use MAX_PATH sized buffer (see https://msdn.microsoft.com/en-us/library/windows/desktop/bb773569(v=vs.85).aspx). It is confusing to change the example to different size.

File: src/pal/src/thread/context.cpp [hunk 3]
```diff
@@ -193,7 +193,7 @@ BOOL CONTEXT_GetRegisters(DWORD processId, ucontext_t *registers)
             return FALSE;
         }
 #elif HAVE_BSD_REGS_T
-        char buf[MAX_PATH];
+        char buf[MAX_LONGPATH];
```
Comment (jkotas) [inline src/pal/src/thread/context.cpp]: This can stay as MAX_PATH

**Comment by Priya91** [comment]
@jkotas Added commits to the PR feedback, please review for final signoff. Thanks!

**Pull request merged**